src/backend/api/meals.js

```diff
@@ -1,15 +1,111 @@
+const { request, response } = require("express");
 const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    let maxPrice = request.query.maxPrice;
+    let availableReservations = request.query.availableReservations;
+    let title = request.query.title;
+    let createdAfter = request.query.createdAfter;
+    let limit = request.query.limit;
+    let limitMaxPrice = maxPrice + limit;
+
+    if (maxPrice) {
+      const totalMaxPrice = await knex("Meal").where("price", "<=",maxPrice);
+      response.json(totalMaxPrice);
+
+    } else if (availableReservations) {
+      //7.Get meals that still has available reservations
+    
+      await knex("Meal")
+        .select(
+          "Meal.id",
+          "Meal.title",
+          "Meal.max_reservations",
+          knex.raw("sum(number_of_guests)")
+        )
+        .innerJoin("Reservation", "Meal.id", "Reservation.meal_id")
+        .groupBy("Meal.id")
+        .having("Meal.max_reservations", ">", "sum(number_of_guests)")
+        .then((data) => response.json(data));
+
+    } else if (title) {
+      const allTitles = await knex("Meal").where("title", "like", title);
+      response.json(allTitles);
+
+    } else if (createdAfter) {
+      const dateCreatedAfter = await knex("Meal").where("created_date",">",createdAfter);
+      response.json(dateCreatedAfter);
+
+    } else if (request.query.limit) {
+      const rLimit = await knex("Meal").limit(request.query.limit);
+      response.json(rLimit);
+
+    } else if (limitMaxPrice) {
+      const rMaxPrice = await knex("Meal").where("price", ">=", maxPrice);
```

Comment:
I am not sure about this solution, but what they asked you to do is combine two or more queries.